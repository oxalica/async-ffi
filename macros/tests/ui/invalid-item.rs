use async_ffi_macros::async_ffi;

#[async_ffi]
pub fn not_async() {}

pub trait Trait {
    #[async_ffi_macros::async_ffi]
    async fn method(&self);
}

impl Trait for () {
    #[async_ffi_macros::async_ffi]
    async fn method(&self) {}
}

fn main() {}
