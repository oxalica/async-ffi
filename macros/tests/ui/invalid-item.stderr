error: #[async_ffi] expects an `async fn`
 --> tests/ui/invalid-item.rs:4:5
  |
4 | pub fn not_async() {}
  |     ^^

error: #[async_ffi] does not support `self` parameter
 --> tests/ui/invalid-item.rs:8:21
  |
8 |     async fn method(&self);
  |                     ^

error: #[async_ffi] does not support `self` parameter
  --> tests/ui/invalid-item.rs:13:21
   |
13 |     async fn method(&self) {}
   |                     ^
